From 81f6d20e344a37aa1d2555b8de6cfb23f7b1f4e8 Mon Sep 17 00:00:00 2001
From: Fujii Hironori <Hironori.Fujii@sony.com>
Date: Fri, 3 Mar 2023 04:15:10 +0900
Subject: [PATCH 15/15] win32 font is very small if the lfHeight of HFONT is
 exactly -size

cairo_win32_font_face_create_for_hfont is reusing the HFONT object
passed by an argument if possible to create a scaled font. However,
the condition was wrong. It checked the font matrix scale factor is
`-lfHeight`. But it should be `-lfHeight * WIN32_FONT_LOGICAL_SCALE`.

Fixes cairo/cairo#3
---
 src/win32/cairo-win32-font.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/win32/cairo-win32-font.c b/src/win32/cairo-win32-font.c
index fd9461363..a561e74a4 100644
--- a/src/win32/cairo-win32-font.c
+++ b/src/win32/cairo-win32-font.c
@@ -1798,7 +1798,7 @@ _cairo_win32_font_face_scaled_font_create (void			*abstract_face,
     if (font_face->hfont) {
         /* Check whether it's OK to go ahead and use the font-face's HFONT. */
         if (_is_scale (ctm, 1.) &&
-            _is_scale (font_matrix, -font_face->logfont.lfHeight)) {
+            _is_scale (font_matrix, -font_face->logfont.lfHeight * WIN32_FONT_LOGICAL_SCALE)) {
             hfont = font_face->hfont;
         }
     }
-- 
2.34.1

